Tests/Microsoft.AppCenter.Test.Windows/Storage/StorageAdapterTest.cs

```diff
@@ -5,63 +5,286 @@
+        }
+
+        [TestMethod]
+        public void FaildToOpenDatabaseWhenNameWrong()
+        {
+            var adapter = new StorageAdapter();
+            try
+            {
+                adapter.Initialize("test://test.txt");
+                Assert.Fail("Should have thrown exception");
+            }
+            catch (Exception e)
+            {
+                Assert.IsTrue(e.Message.Contains("Failed to open database connection"));
+            }
+        }
 
-            // Verify that a directory object was created.
-            Assert.IsNotNull(adapter._databaseDirectory);
+        [TestMethod]
+        public void DatabaseDisposed()
+        {
+            // Prepare data.
+            var adapter = new StorageAdapter();
+            try
+            {
+                adapter.Initialize(DatabasePath);
+            }
+            catch (Exception e)
+            {
+                Assert.Fail("Shouldn't have thrown exception");
+            }
+            try
+            {
+                // Try get data before database initialize.
+                CreateTableHelper(adapter);
+            }
+            catch (Exception e)
+            {
+                Assert.Fail("Shouldn't have thrown exception");
+            }
+            adapter.Dispose();
+            try
+            {
+                // Try get data before database initialize.
+                CreateTableHelper(_adapter);
+                Assert.Fail("Should have thrown exception");
+            }
+            catch (Exception e)
+            {
+                Assert.IsFalse(e.Message.Contains("Should have thrown exception"));
+            }
+        }
 
-            // Replace the directory with a mock and initialize.
-            adapter._databaseDirectory = Mock.Of<Directory>();
-            adapter.InitializeStorageAsync().Wait();
-            Mock.Get(adapter._databaseDirectory).Verify(directory => directory.Create());
+        /// <summary>
+        /// Verify that database is not initilaze.
+        /// </summary>
+        [TestMethod]
+        public void DatabaseIsNotInitilazeWhenCallCount()
+        {
+            // Prepare data.
+            var exception = new StorageException("The database wasn't initialized.");
+            try
+            {
+                // Try get data before database initialize.
+                _adapter.Count(TableName, ColumnChannelName, StorageTestChannelName);
+                Assert.Fail("Should have thrown exception");
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(exception.Message, e.Message);
+            }
+            try
+            {
+                // Initialize database.
+                _adapter.Initialize(DatabasePath);
+            }
+            catch
+            {
+                // Handle exception, database is not created with Mock.
+            }
+            CreateTableHelper(_adapter);
+            try
```

Comment:
What is the difficult outcome of your code and the following code when test executed in CI or automation?

```c#
_adapter.Count(TableName, ColumnChannelName, StorageTestChannelName);
```

At least in Visual Studio Text Explorer, both will fail, not much difference.